src/main/java/frc/robot/GrabberIntake.java

```diff
@@ -1,21 +1,48 @@
 package frc.robot;
-import edu.wpi.first.wpilibj.DoubleSolenoid;
-import edu.wpi.first.wpilibj.PneumaticsModuleType;
 import edu.wpi.first.wpilibj2.command.Subsystem;
-import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.Timer;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
 
 public class GrabberIntake implements Subsystem { 
-    private DoubleSolenoid grabber;
+    private CANSparkMax intakeMotor;
+    private double nomVolt;
+    private Timer timer;
 
     public GrabberIntake(){
-        grabber = new DoubleSolenoid(PneumaticsModuleType.REVPH,0,1);
+        //grabber = new DoubleSolenoid(PneumaticsModuleType.REVPH,0,1);
+        intakeMotor = new CANSparkMax(48,MotorType.kBrushless);
+        intakeMotor.setInverted(true);
+        timer = new Timer();
+        
+    }
+    public void periodic() {
+        SmartDashboard.putNumber("Intake Speed%", intakeMotor.get());
+
     }
 
     public void Grab(boolean grabbing) {
         if (grabbing) {
-            grabber.set(Value.kForward);
+            intakeMotor.setVoltage(Constants.IntakeVoltage);
         } else {
-            grabber.set(Value.kReverse);
+            intakeMotor.setVoltage(-1 * Constants.IntakeVoltage);
+        }
+        nomVolt = intakeMotor.getVoltageCompensationNominalVoltage();
+        SmartDashboard.putNumber("Intake Nominal Voltage", nomVolt);
+        
+        if(nomVolt >= 0) {
+            timer.start();
```

Comment:
You're starting the timer then checking it right after?

Reply:
Yeah the timer needs a little bit of work. What do you suggest? I pushed to git so there would be some framework available.

Reply:
I worked on it, should be ok now